src/Collection.php

```diff
@@ -201,15 +202,18 @@ public function aggregate(array $pipeline, array $options = [])
 
         $server = $this->manager->selectServer($options['readPreference']);
 
-        /* A "majority" read concern is not compatible with the $out stage, so
-         * avoid providing the Collection's read concern if it would conflict.
+        /* Servers before 4.2 don't support a readConcern when an $out stage is
+         * being used. Starting with 4.2, a read concern can be used; however it
+         * does not support the "linearizable" level. Using the "linearizable"
+         * level will cause a server error.
          *
          * A read concern is also not compatible with transactions.
          */
         if ( ! isset($options['readConcern']) &&
-            ! ($hasOutStage && $this->readConcern->getLevel() === ReadConcern::MAJORITY) &&
             \MongoDB\server_supports_feature($server, self::$wireVersionForReadConcern) &&
-            ! \MongoDB\is_in_transaction($options)) {
+            ! \MongoDB\is_in_transaction($options) &&
+            ( ! $hasOutStage || \MongoDB\server_supports_feature($server, self::$wireVersionForReadConcernWithOutStage))
```

Comment:
I think this is also an improvement. IIRC, there was some discussion about venturing a bit too far outside the spec with previous behavior of silently deciding whether or not it was safe to inherit a particular read concern. I'd now say it's preferable to inherit as-is (if supported) and allow the server to throw if it needs to.